fix/solve-merge-issue

giyeon95 opened the pull request:
fix solve merge issue

giyeon95 committed:
Feature/plan api v0.3 (#23)

* 여행 상세페이지 기능 개발 완료

godong9 commented on src/main/java/com/strictmanager/travelbudget/application/member/PaymentVO.java:
```diff
@@ -0,0 +1,29 @@
+import lombok.Getter;
+
+@Getter
+@Builder
```
혹시 PaymentVO 기존거 안쓰고 추가로 만드신 이유 있으실까요~? 
PaymentVO 제가 만들어둔게 있어서...

giyeon95 replied:
GET: /plan/{id} 에서 response 줄때 payment 목록 반환하려 했었습니다! 

client에게 여행 상세 조회시 API를 한번 내려줄지, 여러개로 나눌지에 따라 달라질것 같아요! 

의견주세요~~
 ps) 만약 유지한다면 네이밍은 바꿔야겠네요 :)

godong9 replied:
여행 상세 조회시랑 payment 목록은 분리해서 API 호출하도록 하는게 맞을거같아요~
여행 상세 조회시 Plan, Budget 만 내려주고,
지출내역은 클라에서 budget id 로 호출하도록 하는게 어떨까요?

giyeon95 replied:
넵 좋습니다~

godong9 commented on src/main/java/com/strictmanager/travelbudget/domain/payment/PaymentCase.java:
```diff
@@ -36,6 +36,7 @@ private PaymentCase(
         Long price,
         String title,
         LocalDateTime paymentDt,
+//        LocalDate paymentDate,
```
이거 paymentDt 로 두고 클라에서 필요하다고 하면 Response 에서 분리해서 내려주면 될거같아요~
주석 지우셔도 될듯~

godong9 commented on src/main/java/com/strictmanager/travelbudget/domain/plan/PlanService.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    public List<TripPlan> getDoingPlans(User user) {
+        return tripMemberRepository.findByUserAndTripPlanStartDateBeforeAndTripPlanEndDateGreaterThanEqual(user, LocalDate.now(), LocalDate.now())
```
이거 StartDate 만 비교해도 되지 않을까요~?
모든 플랜은 StartDate 가 EndDate 보다 빨라야된다가 우선 제약조건일거같아요!

giyeon95 replied:
진행중인 여행 조회시 startDate가 현재일보다 작다. 라고 한다면, 종료된 여행을 판단해주지 못할것 같아서 조건을 함께 걸어 줬었어요~ 

과거여행 -> 시작일이 현재일보다 과거, 종료일도 현재일보다 과거
진행중인여행 -> 시작일이 현재보다 과거, 종료일은 현재일보다 미래

godong9 replied:
아 이해했습니다~ 👍

godong9 commented on src/main/java/com/strictmanager/travelbudget/application/member/PlanManager.java:
```diff
@@ -0,0 +1,147 @@
+    public List<PlanResponse> retrievePlans(User user, boolean isComing) {
+
+        Stream<TripPlan> planStream;
+        if (isComing) {
```
이름 직관적이네요 👍

godong9 commented on src/main/java/com/strictmanager/travelbudget/application/member/PlanManager.java:
```diff
@@ -0,0 +1,147 @@
+
+    }
+
+    public void createPlan(PlanVO vo) {
```
Transaction 걸어줘야할 것 같아요.
귀찮아서 명시적으로 설정은 안해줬지만 어노테이션만 붙이면 트랜잭션은 걸릴듯

giyeon95 replied:
좋은의견 감사합니다~ 👍

godong9 commented on src/main/java/com/strictmanager/travelbudget/application/member/PlanManager.java:
```diff
@@ -0,0 +1,147 @@
+
+    private final long INIT_AMOUNT = 0L;
+
+    public List<PlanResponse> retrievePlans(User user, boolean isComing) {
```
getPlans 로 메서드명 지정하면 이상할까요?

giyeon95 replied:
get으로 통일하는걸로 하겠습니다 :)

giyeon95 replied:
Controller 단에서는 get으로 쓰고, application단에서는 retrieve를 유지하면 어떨까요?? (find나, 다른 네이밍도 좋습니다~)

godong9 replied:
application 단과 Controller 레벨에서 메서드명이 굳이 달라야할까요? 
다르게 하고 싶으시면 find 도 좋은데 저는 같아도 상관없을거같아요 😄

godong9 commented on src/main/java/com/strictmanager/travelbudget/application/member/PlanManager.java:
```diff
@@ -0,0 +1,147 @@
+
+    }
+
+    public PlanDetailResponse retrievePlanDetail(User user, Long planId, LocalDate date) {
```
이거도 getPlanDetail 은 어떨까요?

godong9 commented on src/main/java/com/strictmanager/travelbudget/web/PlanController.java:
```diff
@@ -3,78 +3,187 @@
     @Getter
     @ToString
-    private static class PlanCreateRequest {
+    public static class CreatePlanRequest {
```
테스트용으로 필요하시면 굳이 public 으로 바꾸지마시고
디폴트로 두시고
VisibleForTesting 어노테이션 붙여두면 될거같아요~ (요건 아마 다른 라이브러리에 있는거같은데 기억이;;)

giyeon95 replied:
저런 유용한 어노테이션이,, 꿀팁 감사합니다 :)

giyeon95 reviewed (commented):
좋은 의견주신것 반영하고 몇가지 의견 더 남겨두었습니다~

giyeon95 committed:
reflect code review #1

giyeon95 committed:
reflect code review #2

POST: plans/{id}

remove plan some columns
TODO: move to payment(dayUseAmount, paymentCases)

giyeon95 commented:
동현님~ 리뷰주신내용 참고해서 payment 관련 컬럼 제거했어요~
클라단에서 헷갈릴 수도 있을것 같아서.. 검토 되면 서버에 한번 다시 배포할께요~

godong9 commented:
네네~!

godong9 commented:
이슈 없으면 머지부탁드려요~
